setup.py

```diff
@@ -7,50 +7,19 @@
     from distutils.core import setup
 import ast
 
-
+current_dir = os.path.dirname(__file__)
 def read(fname):
-    inf = open(os.path.join(os.path.dirname(__file__), fname))
+    inf = open(os.path.join(current_dir, fname), "rt", encoding="utf-8")
     out = "\n" + inf.read().replace("\r\n", "\n")
     inf.close()
     return out
 
-
-# Do not import `appdirs` yet, lest we import some random version on sys.path.
-for line in read("appdirs.py").splitlines():
-    if line.startswith("__version__"):
-        version = ast.literal_eval(line.split("=", 1)[1].strip())
-        break
+if __name__ == "__main__":
+	# Do not import `appdirs` yet, lest we import some random version on sys.path.
+	for line in read("appdirs.py").splitlines():
+		if line.startswith("__version__"):
+			version = ast.literal_eval(line.split("=", 1)[1].strip())
+			break
```

Comment:
Please notice the inconsistency in indentation here.  I also believe that double blank lines was to separate the lengthy `setup` call from the *preparation* part (sort-of), so in this case it might not be necessary.